process/smartContract/hooks/blockChainHook.go

```diff
@@ -249,6 +281,16 @@ func (bh *BlockChainHookImpl) GetStorageData(accountAddress []byte, index []byte
 	return value, trieDepth, err
 }
 
+func (bh *BlockChainHookImpl) checkMaxReadsCounters() error {
```

Comment:
Bad naming. The function suggest that only a check is being performed but we increment the counter, Also, for future safety reasons, I suggest to use the mutex whenever we are dealing with the counters

refactoring suggestion:
```GO
func (bh *BlockChainHookImpl) processMaxReadsCounters() error {
    if !bh.enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
        return nil
    }

    bh.mutCounters.Lock
    defer bh.mutCounters.Unlock()

    bh.crtNumberOfTrieReads++
    if bh.crtNumberOfTrieReads > bh.maxNumberOfTrieReadsPerTx {
        return process.OutOfAPICalls
    }
}
```

Reply:
done